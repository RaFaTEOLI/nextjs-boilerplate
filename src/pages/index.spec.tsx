import { render, screen } from '@testing-library/react';
import Home from './index';

describe('Home', () => {
  test('should render the heading', () => {
    const { container } = render(<Home />);

    expect(
      screen.getByRole('heading', { name: /Next.js Boilerplate/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render the colors correctly', () => {
    const { container } = render(<Home />);

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' });
  });
});
