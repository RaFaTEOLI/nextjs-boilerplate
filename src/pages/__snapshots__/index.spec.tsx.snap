// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.c1 {
  width: 10rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem, 100%);
}

<main
  class="c0"
>
  <img
    alt="Logo Next.js"
    class="c1"
    src="/img/logo.png"
  />
  <h1
    class="c2"
  >
    Next.js Boilerplate
  </h1>
  <h2
    class="c3"
  >
    TypeScript, ReactJS, NextJS and Styled Components
  </h2>
  <img
    alt="A developer using a computer."
    class="c4"
    src="/img/coding.svg"
  />
</main>
`;
